pykokkos/lib/ufuncs.py

```diff
@@ -798,4 +798,538 @@ def logaddexp(viewA, viewB):
+    out[tid] = tan(view[tid])
+
+@pk.workunit
+def tan_impl_1d_float(tid: int, view: pk.View1D[pk.float], out: pk.View1D[pk.float]):
+    out[tid] = tan(view[tid])
+
+def tan(view):
+    """
+    Element-wise tangent of the view
+
+    Parameters
+    ----------
+    view : pykokkos view
+           Input view.
+
+    Returns
+    -------
+    out : pykokkos view
+           Output view.
+
+    """
+    if str(view.dtype) == "DataType.double":
+        out = pk.View([view.shape[0]], pk.double)
+        pk.parallel_for(view.shape[0], tan_impl_1d_double, view=view, out=out)
+    elif str(view.dtype) == "DataType.float":
+        out = pk.View([view.shape[0]], pk.float)
+        pk.parallel_for(view.shape[0], tan_impl_1d_float, view=view, out=out)
+    return out
+
+@pk.workunit
+def logical_and_impl_1d_double(tid: int, viewA: pk.View1D[pk.double], viewB: pk.View1D[pk.double], out: pk.View1D[pk.uint16]):
+    out[tid] = viewA[tid] and viewB[tid]
+
+@pk.workunit
+def logical_and_impl_1d_float(tid: int, viewA: pk.View1D[pk.float], viewB: pk.View1D[pk.float], out: pk.View1D[pk.uint16]):
+    out[tid] = viewA[tid] and viewB[tid]
+
+
+def logical_and(viewA, viewB):
+    """
+    Return the element-wise truth value of viewA AND viewB.
+
+    Parameters
+    ----------
+    viewA : pykokkos view
+            Input view.
+    viewB : pykokkos view
+            Input view.
+
+    Returns
+    -------
+    out : pykokkos view (uint16)
+           Output view.
+
+    """
+    out = pk.View([viewA.shape[0]], pk.uint16)
+    if str(viewA.dtype) == "DataType.double" and str(viewB.dtype) == "DataType.double":
+        pk.parallel_for(
+            viewA.shape[0],
+            logical_and_impl_1d_double,
+            viewA=viewA,
+            viewB=viewB,
+            out=out)
+
+    elif str(viewA.dtype) == "DataType.float" and str(viewB.dtype) == "DataType.float":
+        pk.parallel_for(
+            viewA.shape[0],
+            logical_and_impl_1d_float,
+            viewA=viewA,
+            viewB=viewB,
+            out=out)
+    else:
+        raise RuntimeError("Incompatible Types")
+    return out
+
+@pk.workunit
+def logical_or_impl_1d_double(tid: int, viewA: pk.View1D[pk.double], viewB: pk.View1D[pk.double], out: pk.View1D[pk.uint16]):
+    out[tid] = viewA[tid] or viewB[tid]
+
+@pk.workunit
+def logical_or_impl_1d_float(tid: int, viewA: pk.View1D[pk.float], viewB: pk.View1D[pk.float], out: pk.View1D[pk.uint16]):
+    out[tid] = viewA[tid] or viewB[tid]
+
+
+def logical_or(viewA, viewB):
+    """
+    Return the element-wise truth value of viewA OR viewB.
+
+    Parameters
+    ----------
+    viewA : pykokkos view
+            Input view.
+    viewB : pykokkos view
+            Input view.
+
+    Returns
+    -------
+    out : pykokkos view (uint16)
+           Output view.
+
+    """
+    out = pk.View([viewA.shape[0]], pk.uint16)
+    if str(viewA.dtype) == "DataType.double" and str(viewB.dtype) == "DataType.double":
```

Comment:
instead of doing all these comparison with strings, we should introduce some global functions like
`type_eq(viewA.dtype, DataType.double)`, then inside the function convert to anything if necessary
@NaderAlAwar

Reply:
Agreed, I opened #80 to track this. I think we can resolve this for this PR.